overlord/assertstate/assertstate.go

```diff
@@ -356,7 +356,31 @@ func RefreshValidationSetAssertions(s *state.State, userID int) error {
 		return nil
 	}
 
-	return bulkRefreshValidationSetAsserts(s, vsets, userID, deviceCtx)
+	if err := bulkRefreshValidationSetAsserts(s, vsets, userID, deviceCtx); err != nil {
+		return err
+	}
+
+	// update validation set tracking state
+	for _, vs := range vsets {
+		if vs.Mode == Monitor && vs.PinnedAt == 0 {
```

Comment:
we should add a TODO about the fact for enforce mode we will have to check that the validation-sets don't conflict and so are useable before moving Current

Reply:
Done.